code/Controllers/CMSMain.php

```diff
@@ -1748,12 +1738,9 @@ public function currentPageID()
     /**
      * Save and Publish page handler
      *
-     * @param array $data
-     * @param Form $form
-     * @return HTTPResponse
      * @throws HTTPResponse_Exception
      */
-    public function save($data, $form)
+    public function save(array $data, Form $form): HTTPResponse
```

Comment:
This method sometimes returns `Security::permissionFailure()` which sometimes returns an array

Reply:
As noted above, it does not return an array